Todo/AppDelegate.swift

```diff
@@ -0,0 +1,420 @@
+
+        puller = database.createPullReplication(kSyncGatewayUrl)
+        puller.continuous = true
+        puller.authenticator = authenticator
+        puller.headers = headers
+
+        NSNotificationCenter.defaultCenter().addObserver(self, selector: "replicationProgress:",
+            name: kCBLReplicationChangeNotification, object: puller)
+
+        pusher.start()
+        puller.start()
+    }
+    
+    func stopReplication() {
+        if !kSyncEnabled {
+            return
+        }
+
+        pusher.stop()
+        NSNotificationCenter.defaultCenter().removeObserver(
+            self, name: kCBLReplicationChangeNotification, object: pusher)
+        
+        puller.stop()
+        NSNotificationCenter.defaultCenter().removeObserver(
+            self, name: kCBLReplicationChangeNotification, object: puller)
+    }
+    
+    func replicationProgress(notification: NSNotification) {
+        if (pusher.status == .Active || puller.status == .Active) {
+            UIApplication.sharedApplication().networkActivityIndicatorVisible = true
+        } else {
+            UIApplication.sharedApplication().networkActivityIndicatorVisible = false
+        }
+        
+        let error = pusher.lastError ?? puller.lastError
+        if (error != syncError) {
+            syncError = error
+            if let errorCode = error?.code {
+                NSLog("Replication Error: %@", error!)
+                if errorCode == 401 {
+                    Ui.showMessageDialog(
+                        onController: self.window!.rootViewController,
+                        withTitle: "Authentication Error",
+                        withMessage:"Your username or password is not correct.",
+                        withError: nil,
+                        onClose: { () -> Void in
+                            self.logout()
+                    })
+                }
+            }
+        }
+    }
+
+    // MARK: - Conflicts Resolution
+
+    override func observeValueForKeyPath(keyPath: String?, ofObject object: AnyObject?,
+        change: [String : AnyObject]?, context: UnsafeMutablePointer<Void>) {
+            if object as? NSObject == conflictsLiveQuery {
+                resolveConflicts(conflictsLiveQuery!.rows)
+            }
+    }
+
+    func startConflictLiveQuery() {
+        conflictsLiveQuery = database.createAllDocumentsQuery().asLiveQuery()
+        conflictsLiveQuery!.allDocsMode = .OnlyConflicts
+        conflictsLiveQuery!.addObserver(self, forKeyPath: "rows", options: .New, context: nil)
+        conflictsLiveQuery!.start()
+    }
+
+    func stopConflictLiveQuery() {
+        conflictsLiveQuery?.removeObserver(self, forKeyPath: "rows")
+        conflictsLiveQuery?.stop()
+        conflictsLiveQuery = nil
+    }
+
+    func resolveConflicts(queryEnum: CBLQueryEnumerator?) {
+        guard let rows = queryEnum else {
+            return
+        }
+
+        for row in rows {
```

Comment:
I've determined that the best way to use a QueryEnumerator in Swift is like `while let row = rows.nextRow() {...}`. Using `nextRow` ensures that `row` will be correctly typed as a CBLQueryRow, not just AnyObject.

Specifically, if you make that change, you can remove the implicit-optionals in the parameters to the resolve methods below.